assets/src/pages/room/components/StreamPlayer/MediaPlayerPeersSection.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC } from "react";
+import { RemotePeer, Track } from "../../hooks/usePeerState";
+import MediaPlayerTile from "./MediaPlayerTile";
+import { MembraneWebRTC, TrackEncoding } from "@membraneframework/membrane-webrtc-js";
+import clsx from "clsx";
+import { StreamSource, TrackType } from "../../../types";
+
+export type TrackWithId = {
+  stream?: MediaStream;
+  trackId?: string;
+  encodingQuality?: TrackEncoding;
+};
+
+export type MediaPlayerTileConfig = {
+  peerId?: string;
+  emoji?: string;
+  flipHorizontally?: boolean;
+  displayName?: string;
+  video: TrackWithId[];
+  audio: TrackWithId[];
+  screenSharing: TrackWithId[];
+  showSimulcast?: boolean;
+  remoteSimulcast?: boolean;
+  streamSource: StreamSource;
+};
+
+const getTracks = (tracks: Track[], type: TrackType): TrackWithId[] =>
+  tracks
+    .filter((track) => track?.metadata?.type === type)
+    .map(
+      (track): TrackWithId => ({
+        stream: track.mediaStream,
+        trackId: track.trackId,
+        encodingQuality: track.encoding,
+      })
+    );
+
+const getCameraStreams = (peers: RemotePeer[], showSimulcast?: boolean): MediaPlayerTileConfig[] => {
+  return peers.map((peer) => {
+    const videoTracks: TrackWithId[] = getTracks(peer.tracks, "camera");
+    const audioTracks: TrackWithId[] = getTracks(peer.tracks, "audio");
+    const screenSharingTracks: TrackWithId[] = getTracks(peer.tracks, "screensharing");
+
+    return {
+      peerId: peer.id,
+      emoji: peer.emoji,
+      displayName: peer.displayName,
+      video: videoTracks,
+      audio: audioTracks,
+      screenSharing: screenSharingTracks,
+      autoplayAudio: true,
+      showSimulcast: showSimulcast,
+      flipHorizontally: false,
+      remoteSimulcast: true,
+      streamSource: "remote",
+    } as MediaPlayerTileConfig;
```

Comment:
remove `as MediaPlayerTileConfig` and specify type in callback function: `... (peer): MediaPlayerTileConfig => ...`